modules/swagger-jaxrs/src/main/java/com/wordnik/swagger/jaxrs/Reader.java

```diff
@@ -70,9 +102,31 @@ public Reader(Swagger swagger, ReaderConfig config) {
     this.config = new DefaultReaderConfig(config);
   }
 
-  public Swagger read(Set<Class<?>> classes) {
+  public synchronized Swagger read(Set<Class<?>> classes) {
+
+    listeners.clear();
+
+    for(Class<?> cls: classes) {
+      if( ReaderListener.class.isAssignableFrom( cls ) && listeners.containsKey( cls )){
```

Comment:
My point was wrong. I didn't realize initially that we don't need to add implementation of the ReaderListener into our REST resources.
So, having this understanding I have a concern about the `read(Class<?> cls)` method. Inside this method method we don't add our listener into the `listeners` map. If we have two subsequent calls of this method it's possible to activate listener for itself only:
```
reader.read(MyListener.class);
reader.read(MyREST.class); // MyREST will be hidden from MyListener
```
As for now we always call this method from `read(Set<Class<?>> classes)`. Perhaps it makes sense to declare  `read(Class<?> cls)` as private or protected.